tools/dds/dds-adapter/lrs-device-controller.cpp

```diff
@@ -491,48 +489,85 @@ lrs_device_controller::lrs_device_controller( rs2::device dev, std::shared_ptr<
-            if( it != _stream_name_to_server.end() )
-            {
-                auto & server = it->second;
-                if( _bridge.is_streaming( server ) )
+    _bridge.on_start_sensor(
+        [this]( std::string const & sensor_name, dds_stream_profiles const & active_profiles )
+        {
+            auto & sensor = _rs_sensors[sensor_name];
+            auto rs2_profiles = get_rs2_profiles( active_profiles );
+            sensor.open( rs2_profiles );
+            sensor.start(
+                [this]( rs2::frame f )
                 {
-                    realdds::topics::image_msg image;
-                    auto data = static_cast< const uint8_t * >( f.get_data() );
-                    image.raw_data.assign( data, data + f.get_data_size() );
-                    image.height = server->get_image_header().height;
-                    image.width = server->get_image_header().width;
-                    image.timestamp  // in sec.nsec
-                        = realdds::dds_time( static_cast< long double >( f.get_timestamp() ) / 1e3 );
+                    auto const stream_profile = f.get_profile();
+                    auto const stream_name = stream_name_from_rs2( stream_profile );
+                    auto it = _stream_name_to_server.find( stream_name );
+                    if( it == _stream_name_to_server.end() )
+                        return;
+
+                    auto & server = it->second;
+                    if( ! _bridge.is_streaming( server ) )
+                        return;
+
+                    auto const stream_type = stream_profile.stream_type();
+                    dds_time const timestamp  // in sec.nsec
+                        ( static_cast< long double >( f.get_timestamp() ) / 1e3 );
+
                     if( auto video = std::dynamic_pointer_cast< realdds::dds_video_stream_server >( server ) )
+                    {
+                        // RS2_STREAM_DEPTH, RS2_STREAM_COLOR, RS2_STREAM_INFRARED
+                        realdds::topics::image_msg image;
+                        auto data = static_cast<const uint8_t *>(f.get_data());
+                        image.raw_data.assign( data, data + f.get_data_size() );
+                        image.height = video->get_image_header().height;
+                        image.width = video->get_image_header().width;
+                        image.timestamp = timestamp;
                         video->publish_image( std::move( image ) );
+                        publish_frame_metadata( f, timestamp );
+                    }
                     else if( auto motion = std::dynamic_pointer_cast< realdds::dds_motion_stream_server >( server ) )
-                        motion->publish_motion( std::move( image ) );
-                    publish_frame_metadata( f, image.timestamp );
-                }
-            }
+                    {
+                        // RS2_STREAM_GYRO, RS2_STREAM_ACCEL
+                        static realdds::topics::imu_msg imu;
```

Comment:
static variables here can be members.

Reply:
Yeah but then you have to #include the imu message, etc.
I chose to split the start() call into two parts and pass the IMU "context" (message plus mutex) in.